src/services/StructuralMetadataUtils.js

```diff
@@ -222,8 +226,12 @@ export default class StructuralMetadataUtils {
     let filteredSpans = allSpans.filter(span => {
       return toMs(beginTime) < toMs(span.begin);
     });
-    // Return whether new end time overlaps the next begin time
-    return toMs(endTime) > toMs(filteredSpans[0].begin);
+    // Return whether new end time overlaps the next begin time, if there are timespans after the current timespan
+    if (filteredSpans.length !== 0) {
+      return toMs(endTime) > toMs(filteredSpans[0].begin);
+    } else {
+      return false;
+    }
```

Comment:
Mind moving the default `return false;` out of the conditional block?   So you can just get rid of the `else { } `.

Reply:
@Dananji Sorry, for this I meant something like the following...so the function still returns.  This is a common modern JS pattern:
```
if ( something here) {
   return conditionalReturnItem;
}
return defaultReturnItem;  // So this is the default return basically.  Don't need to house it in an else {} block.
```